src/repr/src/row/encode.rs

```diff
@@ -1215,6 +1216,73 @@ impl DatumColumnDecoder {
             | DatumColumnDecoder::RecordEmpty(_) => ColumnStatKinds::None,
         }
     }
+
+    fn goodbytes(&self) -> usize {
+        match self {
+            DatumColumnDecoder::Bool(a) => ArrayOrd::Bool(a.clone()).goodbytes(),
+            DatumColumnDecoder::U8(a) => ArrayOrd::UInt8(a.clone()).goodbytes(),
+            DatumColumnDecoder::U16(a) => ArrayOrd::UInt16(a.clone()).goodbytes(),
+            DatumColumnDecoder::U32(a) => ArrayOrd::UInt32(a.clone()).goodbytes(),
+            DatumColumnDecoder::U64(a) => ArrayOrd::UInt64(a.clone()).goodbytes(),
+            DatumColumnDecoder::I16(a) => ArrayOrd::Int16(a.clone()).goodbytes(),
+            DatumColumnDecoder::I32(a) => ArrayOrd::Int32(a.clone()).goodbytes(),
+            DatumColumnDecoder::I64(a) => ArrayOrd::Int64(a.clone()).goodbytes(),
+            DatumColumnDecoder::F32(a) => ArrayOrd::Float32(a.clone()).goodbytes(),
+            DatumColumnDecoder::F64(a) => ArrayOrd::Float64(a.clone()).goodbytes(),
+            DatumColumnDecoder::Numeric(a) => ArrayOrd::Binary(a.clone()).goodbytes(),
+            DatumColumnDecoder::String(a) => ArrayOrd::String(a.clone()).goodbytes(),
+            DatumColumnDecoder::Bytes(a) => ArrayOrd::Binary(a.clone()).goodbytes(),
+            DatumColumnDecoder::Date(a) => ArrayOrd::Int32(a.clone()).goodbytes(),
+            DatumColumnDecoder::Time(a) => ArrayOrd::FixedSizeBinary(a.clone()).goodbytes(),
+            DatumColumnDecoder::Timestamp(a) => ArrayOrd::FixedSizeBinary(a.clone()).goodbytes(),
+            DatumColumnDecoder::TimestampTz(a) => ArrayOrd::FixedSizeBinary(a.clone()).goodbytes(),
+            DatumColumnDecoder::MzTimestamp(a) => ArrayOrd::UInt64(a.clone()).goodbytes(),
+            DatumColumnDecoder::Interval(a) => ArrayOrd::FixedSizeBinary(a.clone()).goodbytes(),
+            DatumColumnDecoder::Uuid(a) => ArrayOrd::FixedSizeBinary(a.clone()).goodbytes(),
+            DatumColumnDecoder::AclItem(a) => ArrayOrd::FixedSizeBinary(a.clone()).goodbytes(),
+            DatumColumnDecoder::MzAclItem(a) => ArrayOrd::Binary(a.clone()).goodbytes(),
+            DatumColumnDecoder::Range(a) => ArrayOrd::Binary(a.clone()).goodbytes(),
+            DatumColumnDecoder::Json(a) => ArrayOrd::String(a.clone()).goodbytes(),
+            DatumColumnDecoder::Array {
+                dim_offsets,
+                dims,
+                val_offsets,
+                vals,
+                nulls,
+            } => {
+                dim_offsets.inner().inner().len()
```

Comment:
We don't include the offsets size in `goodbytes` calculations... see `ArrayOrd` or `DatumEncoder` for examples.

Reply:
Makes sense! They should all be updated now